flycheck.el

```diff
@@ -4646,10 +4668,8 @@ symbols in the command."
           (setq flycheck-temporaries nil)
           ;; Send the buffer to the process on standard input, if enabled
           (when (flycheck-checker-get checker 'standard-input)
-            (save-restriction
-              (widen)
-              (process-send-region process (point-min) (point-max))
-              (process-send-eof process)))
+            (flycheck--process-send-buffer process)
+            (process-send-eof process))
```

Comment:
I think I'd prefer to make `--process-send-buffer` send the EOF as well.  Let's have everything at just one place :)